File: leetcode/Easy/Minimum-Absolute-Difference.java
1class Solution {
2    public List<List<Integer>> minimumAbsDifference(int[] arr) {
3        List<List<Integer>>ans = new ArrayList<>();
4        Arrays.sort(arr);
5        Integer mn = Integer.MAX_VALUE;
6        for(int i = 0 , j = 1 ;i<arr.length-1;++i,++j)
7        {
8            int diff = Math.abs(arr[i] - arr[j]);
9            mn = Math.min(mn,diff);
10        }
11        for(int i = 0 , j = 1 ;i<arr.length-1;++i,++j)
12        {
13            int diff = Math.abs(arr[i] - arr[j]);
14            ArrayList<Integer> list = new ArrayList<>();
15            list.add(arr[i]);
16            list.add(arr[j]);
17            if(diff == mn)
18                ans.add(list);
19
20
21        }
22        
23        return ans;
24        
25    }
26}

File: leetcode/Easy/Minimum-Absolute-Difference.cpp
1class Solution {
2public:
3    vector<vector<int>> minimumAbsDifference(vector<int>& arr) 
4    {
5        vector<vector<int>>ans;
6        sort(arr.begin(),arr.end());
7        int mn = INT_MAX;
8        for(int i=0,j=1;i<arr.size()-1;++j,++i)
9        {
10            mn= min(mn,abs(arr[j] - arr[i]));
11        }
12                for(int i=0,j=1;i<arr.size()-1;++j,++i)
13                    {
14                        if(abs(arr[j] - arr[i]) == mn)
15                        {
16                            ans.push_back({arr[i],arr[j]});
17                        }
18                    }
19
20        return ans;
21
22
23    }
24};